# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libofx/libofx-0.9.1-gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN libofx-0.9.1-orig/lib/ofx_container_transaction.cpp libofx-0.9.1/lib/ofx_container_transaction.cpp
--- libofx-0.9.1-orig/lib/ofx_container_transaction.cpp	2007-11-10 00:13:21.000000000 +0100
+++ libofx-0.9.1/lib/ofx_container_transaction.cpp	2010-10-07 10:35:14.180001191 +0200
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "messages.hh"
 #include "libofx.h"
diff -ruN libofx-0.9.1-orig/lib/ofx_request.cpp libofx-0.9.1/lib/ofx_request.cpp
--- libofx-0.9.1-orig/lib/ofx_request.cpp	2007-10-28 04:01:38.000000000 +0100
+++ libofx-0.9.1/lib/ofx_request.cpp	2010-10-07 10:35:14.180001191 +0200
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstring>
 #include <string>
 #include "messages.hh"
 #include "libofx.h"
diff -ruN libofx-0.9.1-orig/lib/ofx_request_accountinfo.cpp libofx-0.9.1/lib/ofx_request_accountinfo.cpp
--- libofx-0.9.1-orig/lib/ofx_request_accountinfo.cpp	2007-10-28 04:01:38.000000000 +0100
+++ libofx-0.9.1/lib/ofx_request_accountinfo.cpp	2010-10-07 10:35:14.180001191 +0200
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "libofx.h"
 #include "ofx_request_accountinfo.hh"
diff -ruN libofx-0.9.1-orig/lib/ofx_request_statement.cpp libofx-0.9.1/lib/ofx_request_statement.cpp
--- libofx-0.9.1-orig/lib/ofx_request_statement.cpp	2007-10-28 04:01:38.000000000 +0100
+++ libofx-0.9.1/lib/ofx_request_statement.cpp	2010-10-07 10:35:14.180001191 +0200
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "libofx.h"
 #include "ofx_utilities.hh"
diff -ruN libofx-0.9.1-orig/lib/ofx_utilities.cpp libofx-0.9.1/lib/ofx_utilities.cpp
--- libofx-0.9.1-orig/lib/ofx_utilities.cpp	2007-10-27 14:15:58.000000000 +0200
+++ libofx-0.9.1/lib/ofx_utilities.cpp	2010-10-07 10:35:14.184001191 +0200
@@ -21,7 +21,8 @@
 
 #include "ParserEventGeneratorKit.h"
 #include "SGMLApplication.h"
-#include <time.h>
+#include <ctime>
+#include <cstdlib>
 #include <string>
 #include <locale.h>
 #include "messages.hh"
diff -ruN libofx-0.9.1-orig/ofxconnect/ofxconnect.cpp libofx-0.9.1/ofxconnect/ofxconnect.cpp
--- libofx-0.9.1-orig/ofxconnect/ofxconnect.cpp	2007-01-09 09:04:37.000000000 +0100
+++ libofx-0.9.1/ofxconnect/ofxconnect.cpp	2010-10-07 10:35:14.184001191 +0200
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <cstring>
 
 #ifdef HAVE_LIBCURL
 #include <curl/curl.h>
diff -ruN libofx-0.9.1-orig/ofxconnect/ofxpartner.cpp libofx-0.9.1/ofxconnect/ofxpartner.cpp
--- libofx-0.9.1-orig/ofxconnect/ofxpartner.cpp	2006-03-26 06:49:54.000000000 +0200
+++ libofx-0.9.1/ofxconnect/ofxpartner.cpp	2010-10-07 10:35:14.184001191 +0200
@@ -35,6 +35,9 @@
 #include <iostream>
 #include <string>
 #include <vector>
+#include <algorithm>
+#include <cstring>
+
 
 using std::string;
 using std::vector;
diff -ruN libofx-0.9.1-orig/ofxdump/ofxdump.cpp libofx-0.9.1/ofxdump/ofxdump.cpp
--- libofx-0.9.1-orig/ofxdump/ofxdump.cpp	2006-08-02 06:19:22.000000000 +0200
+++ libofx-0.9.1/ofxdump/ofxdump.cpp	2010-10-07 10:35:14.184001191 +0200
@@ -29,6 +29,8 @@
  ***************************************************************************/
 #include <iostream>
 #include <iomanip>
+#include <cstdlib>
+#include <cstring>
 #include <string>
 #include "libofx.h"
 #include <stdio.h>		/* for printf() */
